app/components/reaction_picker/reaction_picker.js

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Image,
+    View,
+    TouchableWithoutFeedback,
+} from 'react-native';
+
+import Emoji from 'app/components/emoji';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+import {hapticFeedback} from 'app/utils/general';
+import addReactionIcon from 'assets/images/icons/reaction.png';
+import {REACTION_PICKER_HEIGHT} from '../../screens/post_options/post_options_utils';
+
+export default class Reaction extends PureComponent {
+    static propTypes = {
+        addReaction: PropTypes.func.isRequired,
+        openReactionScreen: PropTypes.func.isRequired,
+        theme: PropTypes.object.isRequired,
+        recentEmojis: PropTypes.array,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            selectedIndex: -1,
+            defaultEmojis: [
+                'thumbsup',
+                'smiley',
+                'white_check_mark',
+                'heart',
+                'eyes',
+                'raised_hands',
+            ],
+        };
+    }
+
+    handlePress = (emoji, index) => {
+        this.setState({
+            selectedIndex: index,
+        });
+        hapticFeedback();
+        setTimeout(() => {
+            this.props.addReaction(emoji);
+        }, 250);
```

Comment:
```suggestion
        hapticFeedback();
        this.setState({selectedIndex: index}, () => {
            this.props.addReaction(emoji);
        });
```

This way the add reaction gets executed after the state was set

Reply:
Oh you're right this is much better :D 

Thanks for the tips as always